src/transformers/generation/logits_process.py

```diff
@@ -1411,12 +1417,11 @@ def __init__(self, bos_token_id: int):
         self.bos_token_id = bos_token_id
 
     @add_start_docstrings(LOGITS_PROCESSOR_INPUTS_DOCSTRING)
-    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
-        cur_len = input_ids.shape[-1]
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, cur_len: int) -> torch.FloatTensor:
         if cur_len == 1:
-            num_tokens = scores.shape[1]
-            scores[:, [i for i in range(num_tokens) if i != self.bos_token_id]] = -float("inf")
-            scores[:, self.bos_token_id] = 0
+            mask = torch.full_like(scores, -math.inf)
+            mask[:, self.bos_token_id] = 0
+            scores = scores + mask
```

Comment:
Although technically correct (BOS will be selected), it is not fully backwards compatible: previously, `scores[:, self.bos_token_id]` was set to `0`, now it will not be set.

Perhaps `scores = mask`? Or the whole thing could also be something like `scores = torch.where(torch.arange(scores.shape[1]) == self.bos_token_id, 0, -math.inf))`